Sources/AWSSignerV4/credentials.swift

```diff
@@ -0,0 +1,47 @@
+//
+//  credentials.swift
+//  aws-sign
+//
+//  Created by Adam Fowler on 29/08/2019.
+//
+import Foundation
+
+/// Protocol for providing credential details for accessing AWS services
+public protocol Credential {
+    var accessKeyId: String {get}
+    var secretAccessKey: String {get}
+    var sessionToken: String? {get}
+}
+
+/// basic version of Credential where you supply the credentials
+public struct StaticCredential: Credential {
```

Comment:
We have `StaticCredential` as well as `EnvironmentCredential` in AWSSDKCore as well. I'd remove it from here.

Reply:
No we don't. AWSSDKSwiftCore gets those from AWSSignerV4

Reply:
right you are. sorry.